### PR TITLE
fix: Make crate features with global effects optional.

### PR DESCRIPTION
**Pull request by losfair**
The `swc` crate currently has two features that cause issues when used as a library:

- `log/release_max_level_info`. This feature silences `debug` and `trace` logs globally in *all* crates that depend on the `log` crate.
- `swc_node_base` which depends on the `mimalloc-rust` crate and sets a custom allocator. Users of the `swc` crate may want to provide their own allocator, and the global allocator feature requires a nightly compiler.

This pull request makes these two features optional, while leaving them enabled by default to preserve the current behavior.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -26,7 +26,7 @@ version = "0.54.0"
 name = "swc"
 
 [features]
-default = ["es3"]
+default = ["es3", "filter-logs", "custom-allocator"]
```
Comment (kdy1) [inline Cargo.toml:29]: Please remove these features completely and move them to `node/binding`.

File: Cargo.toml [hunk 2]
```diff
@@ -36,6 +36,10 @@ es3 = []
 #
 # This feature will be removed on v2 of node bindings.
 wrong-target = []
+# Enable this feature to disable logs below the `info` level in release mode.
+filter-logs = ["log/release_max_level_info"]
+# Enable custom allocator. This may improve runtime performance but requires a nightly compiler.
+custom-allocator = ["swc_node_base"]
```
Comment (kdy1) [inline Cargo.toml:42]: This should be removed.

File: Cargo.toml [hunk 3]
```diff
@@ -74,7 +78,7 @@ swc_ecma_transforms_base = {version = "0.31.1", path = "./ecmascript/transforms/
 swc_ecma_utils = {version = "0.44.2", path = "./ecmascript/utils"}
 swc_ecma_visit = {version = "0.38.1", path = "./ecmascript/visit"}
 swc_ecmascript = {version = "0.63.1", path = "./ecmascript"}
-swc_node_base = {version = "0.3.0", path = "./node/base"}
+swc_node_base = {version = "0.3.0", path = "./node/base", optional = true}
```
Comment (kdy1) [inline Cargo.toml:81]: Please move this to `dev-dependencies`. It's required to get correct profiling result.

Comment (kdy1) [reply Cargo.toml:81]: Also, please make it not optional.

**Review by kdy1 — CHANGES_REQUESTED** [review]

**Comment by kdy1** [comment]
I'm curious about the usecase. Can you share it?
Initially, I didn't expect any user to depend on `swc` crate.

**Comment by losfair** [comment]
I'm using `swc` to compile TypeScript files to JavaScript as part of the build process in [rusty-workers](https://github.com/losfair/rusty-workers) v2 (the specific branch is WIP and isn't public yet). The usage looks like:

```rust
let compiler = Compiler::new(Arc::new(SourceMap::default()));
// ... prepare source, handlers and opts
let out = compiler.process_js_file(source, &handler, &opts).unwrap();
```

Is there any better ways to do that? It looks like the parser and codegen are there but needs some glueing to put it together so I just depended on `swc` directly.

**Review by kdy1 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**